insights/client/apps/malware_detection/__init__.py

```diff
@@ -702,7 +702,7 @@ def _get_rules(self):
                     logger.error("Unable to download rules from %s: %s", self.rules_location, str(e))
                     exit(constants.sig_kill_bad)
 
-                if 'CERTIFICATE_VERIFY_FAILED' in str(e):
+                if re.search('SSL.*verify.failed', str(e), re.IGNORECASE):
```

Comment:
How about use the exception type (urllib3.exceptions.SSLError) to make judgement instead of string searching?

Reply:
I'm not sure if one of the machines is returning an SSL error though.  These are the different error messages they are getting:

`2023-02-15 09:21:39,243    DEBUG insights.client.apps.malware_detection Unable to download rules from https://m01eassmg0007.nix.m01.no.myatea.net:443/redhat_access/r/insights/platform/malware-detection/v1/test-rule.yar: HTTPSConnectionPool(host='m01eassmg0007.nix.m01.no.myatea.net', port=443): Max retries exceeded with url: /redhat_access/r/insights/platform/malware-detection/v1/test-rule.yar (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:897)'),))
`
... that one is quite clearly an SSLError exception type.

However this one ...
`2023-03-03 09:59:19,419    DEBUG insights.client.apps.malware_detection Unable to download rules from https://m01eassmg0007.nix.m01.no.myatea.net:443/redhat_access/r/insights/platform/malware-detection/v1/test-rule.yar: ("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",)
`
... is less obvious that the exception is of type SSLError.

That's why I went with string pattern matching.  The regex matches both errors and the customer has tested my regex solution and its working for them.  I'm less confident check if the exeception type is an SSLError will work in the second case above.

Reply:
I think the two error are both 'requests.exceptions.SSLError', but I am not sure and can not test this. So let us use the searching method first.